### DISCUSSION TITLE
如何动态添加国际化语言

### DISCUSSION THREAD

**Discussion by MissNanLan** [umijs/umi#8690] [Q&A] [answered]
使用`addLocale`方法，从后端获取locale，动态添加国际化报错，如图所示

![image](https://user-images.githubusercontent.com/22790863/180635514-104b5df9-5247-4116-95a8-2cf551645bba.png)

layout.tsx
```
useEffect(() => {
   // 后台获取locale 接口
   api.getAllLocales().then((res) => {
      // 动态设置国际化语言，res为后台一个键值对的国际化文案对象
     //  https://v3.umijs.org/zh-CN/plugins/plugin-locale#addlocale
      addLocale('en-US', res, {});
    });
  }, []);
```

此需求非常常见，我在社区中搜到了以下问题，均没有很好的答案

[动态增加语言](https://github.com/ant-design/ant-design-pro/issues/6857)

https://github.com/umijs/umi/issues/6737

https://github.com/umijs/umi/issues/3448

**Comment by fz6m** [comment, marked as answer]
由于缺乏足够的信息，请提供 [最小重现](https://stackoverflow.com/help/minimal-reproducible-example) 的 demo 帮助排查问题。
